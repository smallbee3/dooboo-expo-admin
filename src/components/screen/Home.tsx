import * as WebBrowser from 'expo-web-browser';

import { Button, EditText } from 'dooboo-ui';
import {
  HomeSignInEmailMutation,
  HomeSignInEmailMutationResponse,
} from '../../__generated__/HomeSignInEmailMutation.graphql';
import {
  HomeSignInWithFacebookMutation,
  HomeSignInWithFacebookMutationResponse,
} from '../../__generated__/HomeSignInWithFacebookMutation.graphql';
import {
  HomeSignInWithGoogleMutation,
  HomeSignInWithGoogleMutationResponse,
} from '../../__generated__/HomeSignInWithGoogleMutation.graphql';
import {
  IC_ALERT_TRIANGLE,
  IC_INTRO_BACKGROUND,
  SvgFacebook,
  SvgGoogle,
} from '../../utils/icons';
import React, { ReactElement, useEffect, useState } from 'react';
import {
  facebookAppId,
  facebookSecret,
  googleClientId,
  googleSecret,
} from '../../../config';
import { graphql, useMutation } from 'react-relay/hooks';

import { AuthStackNavigationProps } from '../navigation/AuthNavigator';
import CheckBox from '../shared/CheckBox';
import Hr from '../shared/Hr';
import { Input } from '../ui/Input';
import { SocialAuthProvider } from '../../types';
import SocialSignInButton from '../shared/SocialSignInButton';
import { getString } from '../../../STRINGS';
import styled from 'styled-components/native';
import { useAppContext } from '../../providers/AppProvider';
import { useThemeContext } from '@dooboo-ui/theme';
import { validateEmail } from '../../utils/common';

interface Props {
  navigation: AuthStackNavigationProps<'Home'>;
}

const Container = styled.ImageBackground.attrs({
  source: IC_INTRO_BACKGROUND,
  imageStyle: { opacity: 0.3 },
})`
  flex-direction: row;
  width: 100%;
  height: 100%;
  background-color: ${({ theme }): string => theme.overlay};
`;

const MainTitle = styled.Text`
  font-style: normal;
  font-weight: 300;
  font-size: 32px;
  line-height: 47px;

  color: ${({ theme }): string => theme.overlayFont};
`;

const LeftContainer = styled.View`
  flex: 1;
  justify-content: center;
  align-items: flex-start;
  padding-left: 40px;
`;

const RightContainer = styled.View`
  flex: 1;
  justify-content: center;
  align-items: flex-end;
  padding-right: 40px;
`;

const LoginFormWrapper = styled.View`
  width: 384px;
  background-color: ${({ theme }): string => theme.overlayForm};
  border: 1px solid ${({ theme }): string => theme.overlayFormBorder};
  border-radius: 6px;
  padding: 50px 20px;
`;

const LoginFormTitle = styled.Text`
  font-size: 36px;
  font-weight: bold;
  color: ${({ theme }): string => theme.overlayFont};
  text-align: center;
  margin-bottom: 60px;
`;

const LoginFormOptionWrapper = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 22px;
`;

const CheckBoxLabel = styled.Text`
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 21px;

  margin-right: 10px;

  color: ${({ theme }): string => theme.overlayFont};
`;

const ForgetPasswordLinkedText = styled.Text`
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 21px;

  text-decoration-line: underline;

  color: ${({ theme }): string => theme.overlayFont};
`;

const AlertWarning = styled.View`
  flex-direction: row;
  align-items: center;
  height: 40px;
  background-color: #fcede8;
  border-radius: 6px;
  margin-bottom: 10px;
  padding: 10px;
`;

const AlertWarningImage = styled.Image`
  width: 20px;
  height: 20px;
  margin-right: 7px;
`;

const AlertWarningText = styled.Text`
  color: #d84910;
`;

const AlertWarningLink = styled.Text`
  flex: 1;
  text-align: right;
  text-decoration-line: underline;
  color: #d84910;
`;

const SocialLoginWrapper = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-top: 10px;
  margin-bottom: 20px;
`;

const SignUpLinkWrapper = styled.View`
  flex-direction: row;
  justify-content: center;
`;

const SignUpLinkedText = styled.Text`
  color: ${({ theme }): string => theme.primary};
  font-weight: bold;
  text-align: right;
  text-decoration-line: underline;
`;

const StyledText = styled.Text`
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 21px;

  margin-right: 10px;

  color: ${({ theme }): string => theme.overlayFont};
`;

const ErrorText = styled.Text`
  width: 100%;
  text-align: center;
  font-size: 15px;
  color: red;
  margin-top: 10px;
`;

const signInEmail = graphql`
  mutation HomeSignInEmailMutation($email: String!, $password: String!) {
    signInEmail(email: $email, password: $password) {
      token
      user {
        id
        email
        verified
      }
    }
  }
`;

const signInWithFacebook = graphql`
  mutation HomeSignInWithFacebookMutation($accessToken: String!) {
    signInWithFacebook(accessToken: $accessToken) {
      token
      user {
        id
        email
        verified
      }
    }
  }
`;

const signInWithGoogle = graphql`
  mutation HomeSignInWithGoogleMutation($accessToken: String!) {
    signInWithGoogle(accessToken: $accessToken) {
      token
      user {
        id
        email
        verified
      }
    }
  }
`;

const Home: React.FC<Props> = (props): ReactElement => {
  const { theme } = useThemeContext();
  const { setUser } = useAppContext();

  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const [keepLoginState, setKeepLoginState] = useState(false);
  const [error, setError] = useState<boolean>(false);
  const [needVerification, setNeedVerification] = useState<boolean>(false);

  const [commitSignInEmail, isInFlightEmail] = useMutation<
    HomeSignInEmailMutation
  >(signInEmail);
  const [commitSignInWithFacebook, isInFlightWithFacebook] = useMutation<
    HomeSignInWithFacebookMutation
  >(signInWithFacebook);
  const [commitSignInWithGoogle, isInFlightWithGoogle] = useMutation<
    HomeSignInWithGoogleMutation
  >(signInWithGoogle);

  const handleSubmit = (): void => {
    if (!validateEmail(email)) return setError(true);

    setError(false);

    const mutationConfigEmail = {
      variables: {
        email,
        password,
      },
      onCompleted: async (
        response: HomeSignInEmailMutationResponse,
      ): Promise<void> => {
        const { user } = response.signInEmail;

        if (user && !user.verified) {
          setNeedVerification(true);
        }

        if (user && user.verified) {
          setUser(user);
        }
      },
      onError: (error: any): void => {
        console.error(error);
        setError(true);
      },
    };

    commitSignInEmail(mutationConfigEmail);
  };

  const handleSubmitFacebook = (accessToken): void => {
    const mutationConfigFacebook = {
      variables: { accessToken },
      onCompleted: async (
        response: HomeSignInWithFacebookMutationResponse,
      ): Promise<void> => {
        const { user } = response.signInWithFacebook;

        if (user && !user.verified) {
          return setNeedVerification(true);
        }

        if (user && user.verified) {
          return setUser(user);
        }
      },
      onError: (error: any): void => {
        console.error(error);
        setError(true);
      },
    };

    commitSignInWithFacebook(mutationConfigFacebook);
  };

  const handleSubmitGoogle = (accessToken): void => {
    const mutationConfigGoogle = {
      variables: { accessToken },
      onCompleted: async (
        response: HomeSignInWithGoogleMutationResponse,
      ): Promise<void> => {
        const { user } = response.signInWithGoogle;

        if (user && !user.verified) {
          return setNeedVerification(true);
        }

        if (user && user.verified) {
          return setUser(user);
        }
      },
      onError: (error: any): void => {
        console.error(error);
        setError(true);
      },
    };

    commitSignInWithGoogle(mutationConfigGoogle);
  };

  const renderVerificationAlert = (): ReactElement => {
    return (
      <AlertWarning>
        <AlertWarningImage source={IC_ALERT_TRIANGLE} />
        <AlertWarningText>
          {getString('ALERT_NOT_VERIFICATION_EMAIL')}.
        </AlertWarningText>
        <AlertWarningLink
          onPress={(): void =>
            props.navigation.navigate('VerifyEmail', {
              email,
            })
          }
        >
          {getString('VERIFY')}
        </AlertWarningLink>
      </AlertWarning>
    );
  };

  useEffect(() => {
    WebBrowser.maybeCompleteAuthSession();
  }, []);

  return (
    <Container>
      <LeftContainer>
        <MainTitle>{getString('APP_TITLE')}</MainTitle>
        <Button
          testID="login-btn"
          style={{
            borderRadius: 6,
            width: 150,
            height: 40,
            alignItems: 'center',
            marginBottom: 20,
            backgroundColor: theme.overlayForm,
            borderWidth: 1,
            borderStyle: 'solid',
            borderColor: theme.overlayFormBorder,
            marginTop: 20,
          }}
          textStyle={{
            fontSize: 14,
            fontWeight: 'bold',
            color: theme.btnPrimaryFont,
            lineHeight: 21,
            letterSpacing: -0.5,
          }}
          onPress={(): void => props.navigation.navigate('MainBoard')}
          text="둘러보기"
          isLoading={isInFlightEmail}
        />
      </LeftContainer>

      <RightContainer>
        <LoginFormWrapper>
          <LoginFormTitle>{getString('LOGIN')}</LoginFormTitle>

          {needVerification && renderVerificationAlert()}

          <EditText
            testID="input-email"
            style={{
              backgroundColor: 'rgba(255, 255, 255, 0.75)',
              height: 56,
              marginBottom: 10,
            }}
            textStyle={{
              paddingLeft: 15,
              fontSize: 16,
              lineHeight: 24,
              alignItems: 'center',
            }}
            labelTextStyle={{
              fontWeight: 'bold',
              fontSize: 14,
              lineHeight: 21,
            }}
            value={email}
            onChangeText={setEmail}
            placeholder={getString('USERNAME')}
          />

          <EditText
            testID="input-password"
            style={{
              backgroundColor: 'rgba(255, 255, 255, 0.75)',
              height: 56,
              marginBottom: 22,
            }}
            textStyle={{
              paddingLeft: 15,
              fontSize: 16,
              lineHeight: 24,
              alignItems: 'center',
            }}
            labelTextStyle={{
              fontWeight: 'bold',
              fontSize: 14,
              lineHeight: 21,
            }}
            value={password}
            secureTextEntry={true}
            placeholder={getString('PASSWORD')}
            onChangeText={setPassword}
          />

          {error && (
            <ErrorText testID="text-error">
              {getString('ALERT_FAILED_SIGNIN')}
            </ErrorText>
          )}

          <LoginFormOptionWrapper>
            <CheckBox
              checked={keepLoginState}
              onChange={(): void => setKeepLoginState(!keepLoginState)}
            >
              <CheckBoxLabel>{getString('KEEP_LOGIN_STATE')}</CheckBoxLabel>
            </CheckBox>

            <ForgetPasswordLinkedText
              onPress={(): void => props.navigation.navigate('FindPassword')}
            >
              {getString('FORGOT_PASSWORD')}
            </ForgetPasswordLinkedText>
          </LoginFormOptionWrapper>

          <Button
            testID="login-btn"
            style={{
              backgroundColor: theme.btnPrimary,
              borderRadius: 6,
              width: 200,
              height: 48,
              alignItems: 'center',
              marginBottom: 20,
            }}
            textStyle={{
              fontSize: 14,
              fontWeight: 'bold',
              color: theme.btnPrimaryFont,
              lineHeight: 21,
              letterSpacing: -0.5,
            }}
            onPress={handleSubmit}
            text={getString('LOGIN')}
            isLoading={isInFlightEmail}
          />

          <Hr text="or" style={{ height: 21 }} />

          <SocialLoginWrapper>
            <Button
              testID="login-btn"
              style={{
                backgroundColor: theme.googleBackground,
                borderRadius: 6,
                width: 162,
                height: 48,
              }}
              textStyle={{
                fontSize: 14,
                fontWeight: 'bold',
                color: theme.btnPrimaryFont,
                lineHeight: 21,
                letterSpacing: -0.5,
              }}
              leftElement={
                <SvgGoogle
                  width={20}
                  height={20}
                  style={{ marginRight: 6 }}
                  fill="#fff"
                />
              }
              onPress={(): void => console.log('signin with google')}
              text={getString('SIGN_IN_WITH_GOOGLE')}
              isLoading={isInFlightWithGoogle}
            />
            <Button
              testID="login-btn"
              style={{
                backgroundColor: theme.facebookBackground,
                borderRadius: 6,
                width: 162,
                height: 48,
              }}
              textStyle={{
                fontSize: 14,
                fontWeight: 'bold',
                color: theme.btnPrimaryFont,
                lineHeight: 21,
                letterSpacing: -0.5,
              }}
              leftElement={
                <SvgFacebook
                  width={20}
                  height={20}
                  style={{ marginRight: 6 }}
                  fill="#fff"
                />
              }
              onPress={(): void => console.log('signin with facebook')}
              text={getString('SIGN_IN_WITH_FACEBOOK')}
              isLoading={isInFlightWithFacebook}
            />
          </SocialLoginWrapper>

          <SignUpLinkWrapper>
            <StyledText>{getString('NOT_USER_MESSAGE')}</StyledText>
            <SignUpLinkedText
              onPress={(): void => props.navigation.navigate('SignUp')}
            >
              {getString('SIGN_UP')}
            </SignUpLinkedText>
          </SignUpLinkWrapper>
        </LoginFormWrapper>
      </RightContainer>
    </Container>
  );
};

export default Home;
