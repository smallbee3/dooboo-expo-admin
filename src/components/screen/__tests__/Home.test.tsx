import 'react-native';

import React, { ReactElement } from 'react';
import { RenderResult, render } from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import Home from '../Home';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: ReactElement;
let testingLib: RenderResult;

describe('[Home] Rendering', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Home {...props} />);
    testingLib = render(component);
  });

  it('renders without crashing', () => {
    const { baseElement } = testingLib;
    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});
