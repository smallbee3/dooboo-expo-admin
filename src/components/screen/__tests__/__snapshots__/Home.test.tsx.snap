// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Home] Rendering renders without crashing 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "backgroundColor": "#1B1B21",
            "flexDirection": "row",
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            Object {
              "opacity": 0.3,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "paddingLeft": 40,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 32,
                "fontStyle": "normal",
                "fontWeight": "300",
                "lineHeight": 47,
              },
            ]
          }
        >
          Open market integrated solution system
for shopping mall managers
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "paddingRight": 40,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.1)",
                "borderColor": "rgba(255, 255, 255, 0.3)",
                "borderRadius": 6,
                "borderStyle": "solid",
                "borderWidth": 1,
                "paddingBottom": 50,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 50,
                "width": 384,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 36,
                  "fontWeight": "bold",
                  "marginBottom": 60,
                  "textAlign": "center",
                },
              ]
            }
          >
            Login
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "display": "flex",
                  "flexDirection": "column",
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0.75)",
                  "height": 56,
                  "marginBottom": 10,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  Object {
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "lineHeight": 21,
                  },
                  null,
                ]
              }
            />
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              multiline={false}
              placeholder="Username"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "fontSize": 15,
                    "fontWeight": "500",
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                  Object {
                    "alignItems": "center",
                    "fontSize": 16,
                    "lineHeight": 24,
                    "paddingLeft": 15,
                  },
                ]
              }
              testID="input-email"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                  undefined,
                  Object {
                    "borderBottomWidth": 0.6,
                    "borderColor": "#eaeaf9",
                  },
                  null,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "display": "flex",
                  "flexDirection": "column",
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0.75)",
                  "height": 56,
                  "marginBottom": 22,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  Object {
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "lineHeight": 21,
                  },
                  null,
                ]
              }
            />
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              multiline={false}
              placeholder="Password"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "fontSize": 15,
                    "fontWeight": "500",
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                  Object {
                    "alignItems": "center",
                    "fontSize": 16,
                    "lineHeight": 24,
                    "paddingLeft": 15,
                  },
                ]
              }
              testID="input-password"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                  undefined,
                  Object {
                    "borderBottomWidth": 0.6,
                    "borderColor": "#eaeaf9",
                  },
                  null,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 22,
                },
              ]
            }
          >
            <View
              activeOpacity={1}
              reverse={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <View
                  checked={false}
                  color="#609FFF"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#ffffff",
                        "borderColor": "#bababa",
                        "borderRadius": 2,
                        "borderStyle": "solid",
                        "borderWidth": 1,
                        "height": 20,
                        "paddingBottom": 2,
                        "paddingLeft": 2,
                        "paddingRight": 2,
                        "paddingTop": 2,
                        "width": 20,
                      },
                    ]
                  }
                >
                  <Image
                    source={
                      Object {
                        "canInstrument": true,
                        "createTransformer": [Function],
                        "getCacheKey": [Function],
                        "process": [Function],
                      }
                    }
                    style={
                      Array [
                        Object {
                          "height": "100%",
                          "width": "100%",
                        },
                      ]
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "paddingBottom": 5,
                      "paddingLeft": 5,
                      "paddingRight": 5,
                      "paddingTop": 5,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                        "lineHeight": 21,
                        "marginRight": 10,
                      },
                    ]
                  }
                >
                  Keep Login State
                </Text>
              </View>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "lineHeight": 21,
                    "textDecorationLine": "underline",
                  },
                ]
              }
            >
              Do you forget password?
            </Text>
          </View>
          <View
            delayPressIn={30}
            testID="login-btn"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "borderColor": "blue",
                    "flexDirection": "row",
                    "height": 52,
                    "justifyContent": "center",
                    "width": 320,
                  },
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#609FFF",
                    "borderRadius": 6,
                    "height": 48,
                    "marginBottom": 20,
                    "width": 200,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#069ccd",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "white",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "letterSpacing": -0.5,
                      "lineHeight": 21,
                    },
                  ]
                }
              >
                Login
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "height": 21,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#D1D1D1",
                      "height": 1,
                    },
                  ]
                }
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#D1D1D1",
                    "paddingBottom": 5,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "paddingTop": 5,
                  },
                ]
              }
            >
              or
            </Text>
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#D1D1D1",
                      "height": 1,
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 20,
                  "marginTop": 10,
                },
              ]
            }
          >
            <View
              delayPressIn={30}
              testID="login-btn"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "borderColor": "blue",
                      "flexDirection": "row",
                      "height": 52,
                      "justifyContent": "center",
                      "width": 320,
                    },
                    Object {
                      "backgroundColor": "#E04238",
                      "borderRadius": 6,
                      "height": 48,
                      "width": 162,
                    },
                  ]
                }
              >
                <SvgMock
                  fill="#fff"
                  height={20}
                  style={
                    Object {
                      "marginRight": 6,
                    }
                  }
                  width={20}
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#069ccd",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "white",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "letterSpacing": -0.5,
                        "lineHeight": 21,
                      },
                    ]
                  }
                >
                  Google Login
                </Text>
              </View>
            </View>
            <View
              delayPressIn={30}
              testID="login-btn"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "borderColor": "blue",
                      "flexDirection": "row",
                      "height": 52,
                      "justifyContent": "center",
                      "width": 320,
                    },
                    Object {
                      "backgroundColor": "#345997",
                      "borderRadius": 6,
                      "height": 48,
                      "width": 162,
                    },
                  ]
                }
              >
                <SvgMock
                  fill="#fff"
                  height={20}
                  style={
                    Object {
                      "marginRight": 6,
                    }
                  }
                  width={20}
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#069ccd",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "white",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "letterSpacing": -0.5,
                        "lineHeight": 21,
                      },
                    ]
                  }
                >
                  Facebook Login
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "lineHeight": 21,
                    "marginRight": 10,
                  },
                ]
              }
            >
              You don't have Prime's account yet?
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#609FFF",
                    "fontWeight": "bold",
                    "textAlign": "right",
                    "textDecorationLine": "underline",
                  },
                ]
              }
            >
              Sign Up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
